--- a/hw/sparc64/niagara.c
+++ b/hw/sparc64/niagara.c
@@ -163,7 +163,7 @@ static void niagara_machine_class_init(ObjectClass *oc, void *data)
 {
     mc->desc = "Sun4v platform, Niagara";
     mc->init = niagara_init;
-    mc->max_cpus = 1; /* XXX for now */
+    mc->max_cpus = 8;  /* Enable SMP support for Niagara */
     mc->default_boot_order = "c";
     mc->default_cpu_type = SPARC_CPU_TYPE_NAME("Sun-UltraSparc-T1");
     mc->default_ram_id = "sun4v-partition.ram";
